test/worlds/thermal_invalid.sdf

```diff
@@ -0,0 +1,152 @@
+<?xml version="1.0" ?>
+<!--
+  Demo world consisting of a thermal camera sensor and 3 models with
+  different temperatures specified via the thermal system plugin.
+-->
+
+<sdf version="1.6">
+  <world name="thermal_camera">
+    <physics name="1ms" type="ode">
+      <max_step_size>0.001</max_step_size>
+      <real_time_factor>1.0</real_time_factor>
+    </physics>
+    <plugin
+      filename="ignition-gazebo-physics-system"
+      name="ignition::gazebo::systems::Physics">
+    </plugin>
+    <plugin
+      filename="ignition-gazebo-sensors-system"
+      name="ignition::gazebo::systems::Sensors">
+      <render_engine>ogre2</render_engine>
+    </plugin>
+    <plugin
+      filename="ignition-gazebo-scene-broadcaster-system"
+      name="ignition::gazebo::systems::SceneBroadcaster">
+    </plugin>
+
+    <light type="directional" name="sun">
+      <cast_shadows>true</cast_shadows>
+      <pose>0 0 10 0 0 0</pose>
+      <diffuse>0.8 0.8 0.8 1</diffuse>
+      <specular>0.2 0.2 0.2 1</specular>
+      <direction>-0.5 0.1 -0.9</direction>
+    </light>
+
+    <atmosphere type="adiabatic">
+      <temperature>310</temperature>
+    </atmosphere>
+
+    <model name="sphere">
+      <static>true</static>`
+      <pose>0 0.5 0.5 0 0 0</pose>
+      <link name="sphere_link">
+        <collision name="sphere_collision">
+          <geometry>
+            <sphere>
+              <radius>0.5</radius>
+            </sphere>
+          </geometry>
+        </collision>
+
+        <visual name="sphere_visual">
+          <geometry>
+            <sphere>
+              <radius>0.5</radius>
+            </sphere>
+          </geometry>
+          <material>
+            <ambient>0 0 1 1</ambient>
+            <diffuse>0 0 1 1</diffuse>
+            <specular>0 0 1 1</specular>
+          </material>
+          <plugin
+            filename="ignition-gazebo-thermal-system"
+            name="ignition::gazebo::systems::Thermal">
+            <temperature>600.0</temperature>
```

Comment:
Can we change the temperature for the sphere to be below the camera's default minimum temperature value so that we can make sure that values below the minimum are clamped to the minimum (just like how we are checking that values above the max are clamped to the max with the cylinder)?

Reply:
I'm working on adding this test case, but haven't been able to get the results I want yet. I will keep working on it.

Reply:
I've created some draft PRs to show the current status of what I am tring:
* https://github.com/ignitionrobotics/ign-gazebo/pull/621
* https://github.com/ignitionrobotics/ign-rendering/pull/243

Reply:
I think the issue is that ign-rendering does not accept temperature below zero thus giving it ambient temperature.

How about we clamp negative values to 0 in the thermal system (or in ign-rendering) and print out a warning? Theoretically nothing can be colder than absolute zero (0 K) so it's reasonable to do a quick sanity check.

Reply:
I tried clamping negative to 0 in https://github.com/ignitionrobotics/ign-rendering/pull/243, but that didn't seem to work. Perhaps I am doing it too late in the pipeline. I will try this in the gazebo plugin to see if I get better results

Reply:
The PRs above seem to successfully add in this test case now. We will merge the PRs above later since we need to merge this ASAP for an `ign-gazebo` release (merging these PRs would delay the release since it would require a patch release in `ign-rendering`).